### PR TITLE
185072207 improve session expiration checks r82

### PR DESCRIPTION
**Pull request by ttoomey**
https://www.pivotaltracker.com/story/show/185072207
https://github.com/greenriver/hmis-frontend/pull/360

* set X-app-user-id http header on responses to track session expiration
* add dedicated session keep-alive endpoint
* rename user controller to plural to match rails conventions
* exclude settings endpoint from rack attack to avoid devise weirdness
* move json representation of user to method on model, including session duration

### CODE REVIEW

File: drivers/hmis/app/controllers/hmis/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  skip_before_action :authenticate_user!, only: [:show]
+  prepend_before_action :skip_timeout, only: [:show]
+  before_action :clear_etag, only: [:index]
+
+  # Endpoint to retrieve the currently logged-in user.
+  #
+  # This is called by the frontend on initial page load, to determine whether
+  # there is a currently active session.
+  def show
+    render json: (current_hmis_user&.current_user_api_values || {})
```
Comment (gigxz) [inline drivers/hmis/app/controllers/hmis/users_controller.rb:17]: @ttoomey I'm curious why did you decide against putting the user on the app_settings response? Just wondering, I thought you had mentioned that. On the frontend side it looks like we're always requesting them together.

**Review by gigxz — APPROVED** [review]
looks great!

File: drivers/hmis/app/controllers/hmis/users_controller.rb (hunk 1 above)
Comment (ttoomey) [reply drivers/hmis/app/controllers/hmis/users_controller.rb:17]: @gigxz -yes I forgot to mention I reconsidered that. I think keeping them as separate endpoints is cleaner and the performance issues of cascading requests were addressed by making the requests in parallel (and caching). 

On consideration, current user has different cache properties than settings. While they are cached the same way now, we might want to handle them in a service worker or something in the future

Comment (gigxz) [reply drivers/hmis/app/controllers/hmis/users_controller.rb:17]: I see, that makes a lot of sense. Appreciate your thoughtfulness on this!

**Pull request merged**